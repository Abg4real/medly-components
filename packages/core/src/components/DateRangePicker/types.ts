import { HTMLProps, Omit } from '@medly-components/utils';
import { DisplayFormat } from '../DatePicker/types';
import { Placement } from '../Popover/types';

export type DateRangeProps = Omit<HTMLProps<HTMLInputElement>, 'prefix' | 'size' | 'height' | 'value' | 'onChange'> & {
    /** Selected date range */
    value: { startDate: Date | null; endDate: Date | null };
    /** Function to be called on change of the dates */
    onChange: (value: { startDate: Date | null; endDate: Date | null }) => void;
    /** Variants */
    variant?: 'outlined' | 'filled';
    /** Helper Text */
    helperText?: string;
    /** error text */
    errorText?: string;
    /** Size */
    size?: 'S' | 'M';
    /** Takes full width of the parent component */
    fullWidth?: boolean;
    /** Disable date selection */
    disabled?: boolean;
    /** Popover placement */
    popoverPlacement?: Placement;
    /** Min width in px/rem/% (1rem = 10px)*/
    minWidth?: string;
    /** Minimum selectable Date */
    minSelectableDate?: Date;
    /** Maximum selectable Date */
    maxSelectableDate?: Date;
    /** Date display format */
    displayFormat?: DisplayFormat;
    /** From Label */
    startDateLabel?: string;
    /** To Label */
    endDateLabel?: string;
};
