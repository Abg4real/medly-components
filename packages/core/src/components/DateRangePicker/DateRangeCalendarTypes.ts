import { HTMLProps, Omit, WithThemeProp } from '@medly-components/utils';

export interface Props extends Omit<HTMLProps<HTMLInputElement>, 'prefix' | 'size' | 'height' | 'value' | 'onChange'>, WithThemeProp {
    size?: any;
    startMonth?: any;
    endMonth?: any;
    inputId?: any;
    startDate?: any;
    endDate?: any;
    commonCalendarProps?: any;
}
