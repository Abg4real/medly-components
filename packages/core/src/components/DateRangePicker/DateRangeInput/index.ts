export { DateRangeInput as default } from './DateRangeInput';
