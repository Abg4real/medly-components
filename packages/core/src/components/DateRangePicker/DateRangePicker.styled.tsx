import { styled } from '@medly-components/utils';
import { getPosition } from '../Popover/PopoverWrapper/PopoverWrapper.styled';
import { StyledProps } from './types';

export const DateRangePickerStyled = styled('div').attrs(({ theme: { input, text, font } }) => ({ input, text, font }))<StyledProps>`
    & > div:first-child {
        div {
            z-index: 5;
        }
        & > div:last-child {
            top: unset;
            bottom: unset;
            left: unset;
            right: unset;
            ${({ placement }) => getPosition(placement)}
        }
        & > div:first-child {
            width: 100%;
            display: ${({ fullWidth }) => (fullWidth ? 'grid' : 'inline-grid')};
            grid-template-columns: auto 30px auto;
            border: 1px solid ${props => props.input.borderColor};
            border-radius: 5px;
            padding: ${({ theme }) => theme.spacing.S};
            box-sizing: border-box;
            text-align: center;
            background-color: ${({ disabled, input }) => (disabled ? input.disabledBgcolor : input.bgColor)};
            cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'pointer')};
            * {
                background-color: ${({ disabled, input }) => (disabled ? input.disabledBgcolor : input.bgColor)};
                cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'pointer')};
            }

            label {
                text-align: center;
                border: 0;
                input {
                    text-align: center;
                    max-width: 90px;
                    min-height: unset;
                    padding: 0px;
                    font-size: ${({ font, text }) => font.sizes[text.defaults.textSize]};
                    font-weight: ${({ font, text }) => font.weights[text.defaults.textWeight]};
                    color: ${props => props.input.color};
                    background-clip: padding-box;
                    box-sizing: border-box;

                    &:disabled {
                        cursor: not-allowed;
                    }

                    &:focus {
                        border-color: ${props => props.input.focusedBorderColor};
                        outline: 0;
                    }

                    &::placeholder {
                        color: ${props => props.input.placeHolderColor};
                    }

                    ::-webkit-file-upload-button {
                        visibility: hidden;
                    }
                }
            }
        }
    }
`;
