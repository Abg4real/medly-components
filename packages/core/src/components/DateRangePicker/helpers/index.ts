export * from './getFormattedDate';
export * from './getValidDate';
