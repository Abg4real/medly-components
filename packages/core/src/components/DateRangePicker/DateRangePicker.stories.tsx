import { boolean, select } from '@storybook/addon-knobs';
import React, { useState } from 'react';
import DateRangePicker from '.';
import { Placement } from '../Popover/types';
import { DateRangeProps } from './types';

const placement: Placement[] = [
    'top-start',
    'top',
    'top-end',
    'right-start',
    'right',
    'right-end',
    'bottom-start',
    'bottom',
    'bottom-end',
    'left-end',
    'left',
    'left-start'
];

const displayFormats: DateRangeProps['displayFormat'][] = [
    'dd/MM/yyyy',
    'MM/dd/yyyy',
    'dd/yyyy/MM',
    'MM/yyyy/dd',
    'yyyy/dd/MM',
    'yyyy/MM/dd',
    'dd-MM-yyyy',
    'dd-yyyy-MM',
    'MM-dd-yyyy',
    'MM-yyyy-dd',
    'yyyy-dd-MM',
    'yyyy-MM-dd'
];

export const Basic = () => {
    const [dates, setDates] = useState({ startDate: null, endDate: null });

    return (
        <DateRangePicker
            displayFormat={select('Display Format', displayFormats, 'MM/dd/yyyy')}
            required={boolean('Required', false)}
            fromLabel="From"
            toLabel="To"
            id="date-range-picker"
            value={dates}
            onChange={setDates}
            size={select('Size', ['S', 'M'], 'M')}
            popoverPlacement={select('Placement', placement, 'bottom-start')}
            variant={select('Variant', ['filled', 'outlined'], 'filled')}
            fullWidth={boolean('Full Width', false)}
            disabled={boolean('Disabled', false)}
            minSelectableDate={new Date(2020, 1, 1)}
            maxSelectableDate={new Date(2022, 2, 15)}
        />
    );
};
