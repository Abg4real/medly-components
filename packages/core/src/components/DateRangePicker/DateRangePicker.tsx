import { DateRangeInput, OnDatesChangeProps } from '@datepicker-react/styled';
import React, { useState } from 'react';
import FieldWithLabel from '../FieldWithLabel';
import { DateRangePickerStyled } from './DateRangePicker.styled';
import { Props } from './types';

const DateRangePicker: React.SFC<Props> = React.memo(props => {
    const { label, labelPosition, fullWidth, value, onChange, required, disabled, ...restProps } = props;
    const [date, setDate] = useState(value || { startDate: null, endDate: null });
    const [showDatepicker, setShowDatepicker] = useState(null);

    const handleDateChange = ({ startDate, endDate, focusedInput }: OnDatesChangeProps) => {
            setDate({ startDate, endDate });
            setShowDatepicker(focusedInput);
            onChange && onChange({ startDate, endDate });
        },
        handleClick = (event: React.MouseEvent) => {
            event.preventDefault();
            event.stopPropagation();
        };

    return (
        <FieldWithLabel {...{ labelPosition, fullWidth }}>
            {label && (
                <FieldWithLabel.Label showPointer {...{ required, labelPosition }} htmlFor={label}>
                    {label}
                </FieldWithLabel.Label>
            )}
            <DateRangePickerStyled data-testid="react-datepicker" disabled={disabled} onClick={handleClick}>
                <DateRangeInput
                    {...restProps}
                    onDatesChange={handleDateChange}
                    // @ts-ignore
                    showClose={false}
                    onFocusChange={setShowDatepicker}
                    startDate={date.startDate}
                    endDate={date.endDate}
                    focusedInput={showDatepicker}
                    showStartDateCalendarIcon={false}
                    showEndDateCalendarIcon={false}
                />
            </DateRangePickerStyled>
        </FieldWithLabel>
    );
});

DateRangePicker.displayName = 'DateRangePicker';
DateRangePicker.defaultProps = {
    labelPosition: 'left',
    fullWidth: false,
    required: false,
    disabled: false,
    label: ''
};

export default DateRangePicker;
