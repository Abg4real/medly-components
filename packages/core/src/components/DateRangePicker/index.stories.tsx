import { boolean, number, select, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React, { useState } from 'react';
import { Placement } from '../Popover/types';
import { DateRangePicker } from './DateRangePicker';
import { Props } from './types';

const labelPosition: Array<Props['labelPosition']> = ['top', 'bottom', 'left', 'right'];
const placement: Placement[] = [
    'top-start',
    'top',
    'top-end',
    'right-start',
    'right',
    'right-end',
    'bottom-start',
    'bottom',
    'bottom-end',
    'left-end',
    'left',
    'left-start'
];

const DemoComponent = () => {
    const [date, setDate] = useState({ startDate: null, endDate: null });

    return (
        <DateRangePicker
            value={date}
            onChange={setDate}
            placement={select('Placement', placement, 'bottom')}
            minWidth={number('Min Width', 300)}
            disabled={boolean('Disabled', false)}
            label={text('Label', 'Period')}
            labelPosition={select('Label Position', labelPosition, 'left')}
            required={boolean('Required', false)}
        />
    );
};
storiesOf('Core', module).add('DateRangePicker', DemoComponent);
