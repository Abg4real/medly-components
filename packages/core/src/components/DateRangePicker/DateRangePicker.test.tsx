import { render } from '@test-utils';
import React from 'react';
import { DateRangePicker } from './DateRangePicker';

describe('DateRangePicker', () => {
    it('should render properly', () => {
        const { container } = render(
            <DateRangePicker value={{ startDate: new Date(2010, 0, 1), endDate: new Date(2010, 0, 2) }} onChange={jest.fn()} />
        );
        expect(container).toMatchSnapshot();
    });
});
