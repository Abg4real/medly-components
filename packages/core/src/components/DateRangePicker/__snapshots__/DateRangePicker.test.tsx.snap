// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateRangePicker should render properly 1`] = `
.c13 {
  font-size: 1.2rem;
  line-height: 1.6rem;
  margin: 0.5rem 1.6rem 0;
  white-space: initial;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c10 {
  top: 50%;
  left: 0;
  cursor: text;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: absolute;
  pointer-events: none;
  -webkit-transition: all 150ms cubic-bezier(0.4,0,0.2,1);
  transition: all 150ms cubic-bezier(0.4,0,0.2,1);
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  font-family: Open Sans,Helvetica Neue,Helvetica,Arial,sans-serif;
  -webkit-transform-origin: 0 0;
  -ms-transform-origin: 0 0;
  transform-origin: 0 0;
  touch-action: manipulation;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  opacity: 1;
  z-index: 1;
  width: 100%;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 100%;
}

.c8 {
  position: absolute;
  left: 0;
  opacity: 0;
  cursor: text;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  pointer-events: none;
  color: rgba(0,90,238,.2);
  bottom: 0.7rem;
}

.c0 {
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
  min-width: 33.8rem;
  margin: 0.8rem 0.8rem 0.8rem 0;
}

.c6 {
  color: #13181D;
  width: 100%;
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  background: transparent;
  border: none;
  text-overflow: ellipsis;
  resize: none;
  z-index: 1;
  -webkit-scroll-padding-top: 10px;
  -moz-scroll-padding-top: 10px;
  -ms-scroll-padding-top: 10px;
  scroll-padding-top: 10px;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  padding-bottom: 0.7rem;
}

.c6,
.c6 ~ .c7 {
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  font-family: Open Sans,Helvetica Neue,Helvetica,Arial,sans-serif;
}

.c6:focus {
  outline: none;
}

.c6:focus::-webkit-input-placeholder {
  opacity: 1;
}

.c6:focus::-moz-placeholder {
  opacity: 1;
}

.c6:focus:-ms-input-placeholder {
  opacity: 1;
}

.c6:focus::placeholder {
  opacity: 1;
}

.c6:disabled ~ .c9,
.c6:disabled ~ .c7 {
  cursor: not-allowed;
}

.c6::-webkit-input-placeholder {
  opacity: 0;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
}

.c6::-moz-placeholder {
  opacity: 0;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
}

.c6:-ms-input-placeholder {
  opacity: 0;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
}

.c6::placeholder {
  opacity: 0;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
}

.c6:not(:placeholder-shown) ~ .c7 {
  opacity: 1;
}

.c3 {
  overflow: visible;
  font-size: 2.4rem;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  cursor: pointer;
}

.c3 * {
  fill-opacity: 1;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  fill: #607890;
}

.c4 {
  cursor: pointer;
  border-radius: 50%;
  padding: 0.8rem;
}

.c4 * {
  fill: #607890;
}

.c4:hover {
  background-color: #dfe4e9;
}

.c4:hover * {
  fill: #303C48;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  height: 5.6rem;
  padding: 0 1.6rem;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  cursor: text;
  border-radius: 0.4rem 0.4rem 0 0;
  background-color: #eff2f4;
  padding-left: 0.8rem;
}

.c1 .c9 {
  color: #546A7F;
}

.c1 .c14,
.c1 .sc-AxhCb {
  color: #546A7F;
}

.c1 .c14 *,
.c1 .sc-AxhCb * {
  fill: #546A7F;
}

.c1 ~ .c12 {
  color: #546A7F;
}

.c1::after {
  content: '';
  width: 100%;
  height: 0;
  position: absolute;
  bottom: 0;
  left: 0;
  box-sizing: border-box;
  border-top: 0.1rem solid #546A7F;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
}

.c1:hover::after,
.c1:focus-within::after {
  border-width: 0.2rem;
}

.c1:focus-within,
.c1:focus-within:hover {
  background-color: #E7F0FF;
}

.c1:focus-within::after,
.c1:focus-within:hover::after {
  border-color: #126AFA;
}

.c1:focus-within .c9,
.c1:focus-within:hover .c9 {
  color: #126AFA;
}

.c1:focus-within .c14,
.c1:focus-within:hover .c14,
.c1:focus-within .sc-AxhCb,
.c1:focus-within:hover .sc-AxhCb {
  color: #126AFA;
}

.c1:focus-within .c14 *,
.c1:focus-within:hover .c14 *,
.c1:focus-within .sc-AxhCb *,
.c1:focus-within:hover .sc-AxhCb * {
  fill: #126AFA;
}

.c1:focus-within input::-webkit-input-placeholder,
.c1:focus-within:hover input::-webkit-input-placeholder {
  color: rgba(0,90,238,.2);
}

.c1:focus-within input::-moz-placeholder,
.c1:focus-within:hover input::-moz-placeholder {
  color: rgba(0,90,238,.2);
}

.c1:focus-within input:-ms-input-placeholder,
.c1:focus-within:hover input:-ms-input-placeholder {
  color: rgba(0,90,238,.2);
}

.c1:focus-within input::placeholder,
.c1:focus-within:hover input::placeholder {
  color: rgba(0,90,238,.2);
}

.c1 .c2 {
  margin-right: 0.8rem;
}

.c11 {
  border-right: 0.1rem solid rgba(84,106,127,0.2);
  height: 3.2rem;
  margin-right: 1.6rem;
  border-radius: 0.1rem;
}

<div>
  <div
    class="c0"
    id="medly-date-range-picker-wrapper"
  >
    <div
      class="sc-AxhUy c1"
      id="medly-date-range-picker-textFields"
    >
      <svg
        class="c2 c3 c4"
        fill="none"
        height="1em"
        id="medly-date-range-picker-calendar-icon"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M19 4h-1V3c0-.55-.45-1-1-1s-1 .45-1 1v1H8V3c0-.55-.45-1-1-1s-1 .45-1 1v1H5c-1.11 0-1.99.9-1.99 2L3 20a2 2 0 0 0 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 15c0 .55-.45 1-1 1H6c-.55 0-1-.45-1-1V9h14v10zM9 13v-2H7v2h2zm2-2h2v2h-2v-2zm6 2v-2h-2v2h2z"
          fill="#000"
          fill-opacity=".54"
          fill-rule="evenodd"
        />
      </svg>
      <div
        class="c5"
        id="medly-date-range-picker-from-input-wrapper"
      >
        <input
          class="c6"
          id="medly-date-range-picker-from-input"
          name="START_DATE"
          placeholder="MM - DD - YYYY"
          value="01 - 01 - 2010"
        />
        <span
          class="c7 c8"
        >
          01 - 01 - 2010
        </span>
        <label
          class="c9 c10"
          for="medly-date-range-picker-from-input"
        >
          From
        </label>
      </div>
      <i
        class="c11"
      />
      <div
        class="c5"
        id="medly-date-range-picker-to-input-wrapper"
      >
        <input
          class="c6"
          id="medly-date-range-picker-to-input"
          name="END_DATE"
          placeholder="MM - DD - YYYY"
          value="01 - 02 - 2010"
        />
        <span
          class="c7 c8"
        >
          01 - 02 - 2010
        </span>
        <label
          class="c9 c10"
          for="medly-date-range-picker-to-input"
        >
          To
        </label>
      </div>
    </div>
    <span
      class="c12 c13"
      id="medly-date-range-picker-helper-text"
    />
  </div>
</div>
`;
